core/src/modules/configLoaders/index.ts

```diff
@@ -1,39 +1,64 @@
-import { log, config, env, api } from "actionhero";
-import path from "path";
+import { config, env, log } from "actionhero";
 import fs from "fs";
 import glob from "glob";
 import JSON5 from "json5";
+import path from "path";
+import Sequelize from "sequelize";
 import {
   AnyConfigurationObject,
-  sortConfigurationObjects,
-  validateConfigObjects,
-  IdsByClass,
-  getDirectParentId,
-  SettingConfigurationObject,
+  ApiKeyConfigurationObject,
   AppConfigurationObject,
-  SourceConfigurationObject,
-  PropertyConfigurationObject,
+  DestinationConfigurationObject,
+  getDirectParentId,
   GroupConfigurationObject,
+  IdsByClass,
+  PropertyConfigurationObject,
   ScheduleConfigurationObject,
-  DestinationConfigurationObject,
-  ApiKeyConfigurationObject,
+  SettingConfigurationObject,
+  sortConfigurationObjects,
+  SourceConfigurationObject,
   TeamConfigurationObject,
   TeamMemberConfigurationObject,
+  validateConfigObjects,
 } from "../../classes/codeConfig";
 import { GrouparooErrorSerializer } from "../../config/errors";
-import { loadApp, deleteApps } from "./app";
-import { loadSource, deleteSources } from "./source";
-import { loadProperty, deleteProperties } from "./property";
-import { loadApiKey, deleteApiKeys } from "./apiKey";
-import { loadTeam, deleteTeams } from "./team";
-import { loadTeamMember, deleteTeamMembers } from "./teamMember";
-import { loadGroup, deleteGroups } from "./group";
-import { loadSchedule, deleteSchedules } from "./schedule";
-import { loadSetting } from "./setting";
-import { loadDestination, deleteDestinations } from "./destination";
 import { ConfigWriter } from "../configWriter";
+import { deleteApiKeys, loadApiKey } from "./apiKey";
+import { deleteApps, loadApp } from "./app";
+import { deleteDestinations, loadDestination } from "./destination";
+import { deleteGroups, loadGroup } from "./group";
 import { loadProfile } from "./profile";
-import Sequelize from "sequelize";
+import { deleteProperties, loadProperty } from "./property";
+import { deleteSchedules, loadSchedule } from "./schedule";
+import { loadSetting } from "./setting";
+import { deleteSources, loadSource } from "./source";
+import { deleteTeams, loadTeam } from "./team";
+import { deleteTeamMembers, loadTeamMember } from "./teamMember";
+
+export function getSeenIds(configObjects: AnyConfigurationObject[]) {
+  return configObjects.reduce(
+    (agg, co) => {
+      if (co.id) {
+        const className = co.class.toLowerCase();
+        agg[className] = agg[className] || [];
+        agg[className].push(co.id);
+      }
+      return agg;
+    },
+    {
+      app: [],
+      source: [],
+      property: [],
+      group: [],
+      schedule: [],
+      destination: [],
+      apikey: [],
+      team: [],
+      teammember: [],
+      profile: [],
```

Comment:
Need all these defaults because tests got angry about `undefined` even though the types allow for it.

Reply:
Probably not worth the time to update the tests.